[WIP] Fix documentation of default values in decomposition._dict_learning

clmbst opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs
See #15761


#### What does this implement/fix? Explain your changes.
This PR changes the optional parameters to default in decomposition.dict_learning classes

Commit:
Fix doc of default value in class of decomposition._dict_learning

Commit:
Merge remote-tracking branch 'upstream/master' into default_doc_decomposition

synchronize with updtream before PR

amueller commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -46,14 +46,14 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
         Precomputed covariance, dictionary * X'
 
-    algorithm : {'lasso_lars', 'lasso_cd', 'lars', 'omp', 'threshold'}
+    algorithm : {'lasso_lars', 'lasso_cd', 'lars', 'omp', 'threshold'}, default='lasso_lars'
```
Thanks for the PR! You need to break long lines with ``\``, see the linting errors.

clmbst committed:
Merge remote-tracking branch 'upstream/master' into default_doc_decomposition

clmbst committed:
correct line length

clmbst committed:
TEST

clmbst committed:
undo TEST

clmbst commented:
Thank you for your quick feedback and sorry for such basic misstake.

adrinjalali commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -239,31 +241,31 @@ def sparse_encode(X, dictionary, *, gram=None, cov=None,
         `lasso_lars`.
 
-    n_jobs : int or None, optional (default=None)
+    n_jobs : int or None, default=None
```
if the default is None, it doesn't have to be listed, i.e. you could write `: int, default=None`

adrinjalali commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -63,30 +64,30 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
         from the projection dictionary * data'
 
-    regularization : int | float
+    regularization : int | float, default=None
```
I think we usually say `: int or float, default=None` in this case.

clmbst replied:
What should be done with cases like line 230 where the default is None which means in fact 0.1 * n_features ? I have been inconsistent with those cases through the script. 

`n_nonzero_coefs` : int, 0.1 * n_features by default`

adrinjalali reviewed (commented):
Thanks for the PR @clmbst , and there's no need to apologize for small mistakes or typos. We all do it :)

clmbst committed:
:q

:s
Merge remote-tracking branch 'upstream/master' into default_doc_decomposition

clmbst committed:
fix some doc strings

clmbst commented:
What should be done with cases like line 230 where the default is None which means in fact 0.1 * n_features ? I have been inconsistent with those cases through the script.

n_nonzero_coefs : int, 0.1 * n_features by default`

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -46,14 +46,15 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
 
-    algorithm : {'lasso_lars', 'lasso_cd', 'lars', 'omp', 'threshold'}
+    algorithm : {'lasso_lars', 'lasso_cd', 'lars', 'omp', 'threshold'}, \
+    default='lasso_lars'
```
```suggestion
            default='lasso_lars'
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -63,30 +64,30 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
         `lasso_lars`.
 
-    copy_cov : boolean, optional
+    copy_cov : boolean, default=True
```
```suggestion
    copy_cov : bool, default=True
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -63,30 +64,30 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
         overwritten.
 
-    check_input : boolean, optional
+    check_input : boolean, default=True
```
```suggestion
    check_input : bool, default=True
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -209,13 +210,14 @@ def sparse_encode(X, dictionary, *, gram=None, cov=None,
 
-    algorithm : {'lasso_lars', 'lasso_cd', 'lars', 'omp', 'threshold'}
+    algorithm : {'lasso_lars', 'lasso_cd', 'lars', 'omp', 'threshold'}, \
+    default='lasso_lars'
```
```suggestion
            default='lasso_lars'
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -239,31 +241,31 @@ def sparse_encode(X, dictionary, *, gram=None, cov=None,
         `n_nonzero_coefs`.
 
-    copy_cov : boolean, optional
+    copy_cov : boolean, default=True
```
```suggestion
    copy_cov : bool, default=True
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -239,31 +241,31 @@ def sparse_encode(X, dictionary, *, gram=None, cov=None,
         overwritten.
 
-    init : array of shape (n_samples, n_components)
+    init : array of shape (n_samples, n_components), default=None
```
```suggestion
    init : ndarray of shape (n_samples, n_components), default=None
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -209,13 +210,14 @@ def sparse_encode(X, dictionary, *, gram=None, cov=None,
         output.
 
-    gram : array, shape=(n_components, n_components)
+    gram : array of shape (n_components, n_components), default=None
```
```suggestion
    gram : ndarray of shape (n_components, n_components), default=None
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -209,13 +210,14 @@ def sparse_encode(X, dictionary, *, gram=None, cov=None,
         Precomputed Gram matrix, dictionary * dictionary'
 
-    cov : array, shape=(n_components, n_samples)
+    cov : array of shape (n_components, n_samples), default=None
```
```suggestion
    cov : ndarray of shape (n_components, n_samples), default=None
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -63,30 +64,30 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
         Otherwise it corresponds to n_nonzero_coefs.
 
-    init : array of shape (n_samples, n_components)
+    init : array of shape (n_samples, n_components), default=None
```
```suggestion
    init : ndarray of shape (n_samples, n_components), default=None
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -46,14 +46,15 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
         gram can be None if method is 'threshold'.
 
-    cov : array, shape=(n_components, n_samples)
+    cov : array of shape (n_components, n_samples), default=None
```
```suggestion
    cov : ndarray of shape (n_components, n_samples), default=None
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -63,30 +64,30 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
         Controls the verbosity; the higher, the more messages. Defaults to 0.
 
-    positive: boolean
+    positive: boolean, default=False
```
```suggestion
    positive: bool, default=False
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -239,31 +241,31 @@ def sparse_encode(X, dictionary, *, gram=None, cov=None,
         for more details.
 
-    check_input : boolean, optional
+    check_input : boolean, default=True
```
```suggestion
    check_input : bool, default=True
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -239,31 +241,31 @@ def sparse_encode(X, dictionary, *, gram=None, cov=None,
         Controls the verbosity; the higher, the more messages. Defaults to 0.
 
-    positive : boolean, optional
+    positive : boolean, default=False
```
```suggestion
    positive : bool, default=False
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -446,62 +448,62 @@ def dict_learning(X, n_components, *, alpha, max_iter=100, tol=1e-8,
         for more details.
 
-    dict_init : array of shape (n_components, n_features),
+    dict_init : array of shape (n_components, n_features), default=None
```
```suggestion
    dict_init : ndarray of shape (n_components, n_features), default=None
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -446,62 +448,62 @@ def dict_learning(X, n_components, *, alpha, max_iter=100, tol=1e-8,
         Initial value for the dictionary for warm restart scenarios.
 
-    code_init : array of shape (n_samples, n_components),
+    code_init : array of shape (n_samples, n_components), default=None
```
```suggestion
    code_init : ndarray of shape (n_samples, n_components), default=None
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -646,84 +648,84 @@ def dict_learning_online(X, n_components=2, *, alpha=1, n_iter=100,
         Whether to also return the code U or just the dictionary V.
 
-    dict_init : array of shape (n_components, n_features),
+    dict_init : array of shape (n_components, n_features), default = None
```
```suggestion
    dict_init : ndarray of shape (n_components, n_features), default = None
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -949,7 +951,7 @@ class SparseCoder(SparseCodingMixin, BaseEstimator):
     Parameters
     ----------
-    dictionary : array, [n_components, n_features]
+    dictionary : array of shape (n_components, n_features)
```
```suggestion
    dictionary : ndarray of shape (n_components, n_features)
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -1316,39 +1318,39 @@ class MiniBatchDictionaryLearning(SparseCodingMixin, BaseEstimator):
         whether to shuffle the samples before forming batches
 
-    dict_init : array of shape (n_components, n_features),
+    dict_init : array of shape (n_components, n_features), default=None
```
```suggestion
    dict_init : ndarray of shape (n_components, n_features), default=None
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -1316,39 +1318,39 @@ class MiniBatchDictionaryLearning(SparseCodingMixin, BaseEstimator):
 
     transform_algorithm : {'lasso_lars', 'lasso_cd', 'lars', 'omp', \
-    'threshold'}
+    'threshold'}, default='omp'
```
```suggestion
            'threshold'}, default='omp'
```

glemaitre commented on sklearn/decomposition/_dict_learning.py:
```diff
@@ -1373,40 +1375,40 @@ class MiniBatchDictionaryLearning(SparseCodingMixin, BaseEstimator):
     Attributes
     ----------
-    components_ : array, [n_components, n_features]
+    components_ : array of shape (n_components, n_features)
```
```suggestion
    components_ : ndarray of shape (n_components, n_features)
```

glemaitre reviewed (commented):
A couple of changes to follow our documentation guideline

clmbst committed:
Merge remote-tracking branch 'upstream/master' into default_doc_decomposition

clmbst committed:
doc fix array to ndarray when relevent, indent fix, boolean to bool

glemaitre commented:
Thanks @clmbst
